include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1821,9 +1821,7 @@ auto
 __parallel_stable_sort(oneapi::dpl::__internal::__device_backend_tag __backend_tag, _ExecutionPolicy&& __exec,
                        _Range&& __rng, _Compare __comp, _Proj __proj)
 {
-    auto __cmp_f = [__comp, __proj](const auto& __a, const auto& __b) mutable {
-        return __comp(__proj(__a), __proj(__b));
-    };
+    auto __cmp_f = oneapi::dpl::__internal::__compare<_Compare, _Proj>{__comp, __proj};
     return __parallel_sort_impl(__backend_tag, ::std::forward<_ExecutionPolicy>(__exec), ::std::forward<_Range>(__rng),
                                 __cmp_f);
```

Comment:
Could we pass the object by move here (i.e., `std::move(__cmp_f)`) or is there a downside to that?

Reply:
good question...
In my opinion,  passing an object by move makes sense just for the stateful functors. Due to in the parallel calls the stateless functors are used, usage of `std::move` just makes the code more complicated and less readable... But, I open to discuss it.

Reply:
I would instead create `__internal::__compare` directly as a temporary to pass to the function (i.e. without assigning to a variable).

Reply:
Ready to improve with this suggested change.